core/src/main/kotlin/com/malinskiy/marathon/execution/queue/TestResultReporter.kt

```diff
@@ -57,14 +68,21 @@ class TestResultReporter(
         }
         state<TestState.Executed> {
             on<TestEvent.Failed> {
-                if (configuration.strictMode || count <= 1) {
+                if (count <= 1) {
                     transitionTo(TestState.Failed(it.device, it.testResult), TestAction.SaveReport(it.device, it.testResult))
                 } else {
-                    transitionTo(TestState.Executed(it.device, it.testResult, count - 1))
+                    when (configuration.strictMode) {
+                        StrictMode.ANY_SUCCESS ->
+                            transitionTo(TestState.Executed(it.device, it.testResult, count - 1))
+                        StrictMode.ALL_SUCCESS,
+                        StrictMode.ANY_FAIL ->
+                            transitionTo(TestState.Failed(it.device, it.testResult), TestAction.SaveReport(it.device, it.testResult))
+                    }
                 }
             }
             on<TestEvent.Incomplete> {
                 if (it.final) {
+                    // TODO
```

Comment:
How should we react to Incomplete test when ANY_FAIL was set?

Reply:
We should restart it since it might be a device problem. The final flag is indicating if we reached uncompleted retry quota for example

Reply:
So, I could change nothing in this state